### DISCUSSION TITLE
Workflow_dispatch - Is it possible to pass a secret as parameter?

### DISCUSSION THREAD

**Discussion by gderaco** [community/community#26748] [Actions]
<p>Hi and thanks for the new feature, it was really important to have it as part of GitHub Actions.<br>
I would like to know if it is possible to pass a secret as a parameter (inputs).</p>
<p>Thanks.</p>

**Comment by dogboydog** [comment]
<p>Hi, I’m not sure if I can help or not but do you mean passing the name of a secret, or passing the value and treating it as a secret and masking it?</p>
<p>For the former, maybe there’s a way you can build an expression to fetch the secret based on the input. I don’t know the exact syntax at the moment. Maybe like:</p>
<pre><code class="lang-yaml">inputs:
  myVarSecretName:
    description: 'name of the secret'
    required: true
[...]
    env:
      MY_ENV_VAR: ${{ secrets.$INPUT_MYVARSECRETNAME }}
</code></pre>
<p>Again, I don’t know if this exact syntax will work, I am new to workflow coding.</p>
<p>Edit: This syntax doesn’t work. Trying other versions</p>
<p>For the latter, there’s this <a href="https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#masking-a-value-in-log" rel="nofollow noopener">https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#masking-a-value-in-log</a></p>

**Comment by dogboydog** [comment]
<p>Alright I think I got you. Assuming you meant the first interpretation from my other post.</p>
<p>I can’t fully test since I can’t seem to run any manual workflows, but I tested this using an env variable.</p>
<p>Do this on each step that you need the secret, as apparently <a href="https://stackoverflow.com/questions/61255989/dynamically-retrieve-github-actions-secret" rel="nofollow noopener">you can’t access the “env” context from the workflow-wide “env” block at the top of the file.</a></p>
<pre><code class="lang-yaml">        env:
          MY_ENV_VAR: ${{secrets[env.INPUT_MYSECRETNAME]}} # use auto-generated env var for input
</code></pre>
<p>or possibly:</p>
<pre><code class="lang-yaml">        env:
          MY_ENV_VAR: ${{secrets[inputs.mySecretName]}} # use the ID for the input to access its value
</code></pre>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/gderaco">@gderaco</a>,</p>
<p>It’s possible to pass a secret(for example: PAT) as input to the manual workflow.<br>
Please check my workflow for your reference. I passed a PAT and use it to push the change to the repository, it’s successful.<br>
</p><aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/weide-zhou/ticket14/actions/runs/162743472/workflow" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/52772761/181099734-d80495da-9254-4095-8c01-3b303ae4dfed.png" class="thumbnail onebox-avatar" width="420" height="420">

<h3><a href="https://github.com/weide-zhou/ticket14/actions/runs/162743472/workflow" target="_blank" rel="nofollow noopener">weide-zhou/ticket14</a></h3>

<p>Contribute to weide-zhou/ticket14 development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Note: check the log, the secret value is not protected, could cause data leaking.</p>
<p>Thanks.</p>

**Comment by gderaco** [comment, marked as answer]
<p>Thanks.<br>
This is what worked for me</p>
<p><code>INJECTED_ENV_VAR:     ${{secrets[github.event.inputs.injectedEnvVar]}}</code></p>

**Reply by ecout** [reply]
add_mask does the equivalent, you can still access the value if you have write access to the workflow files but this is an improvement.

**Comment by amithm7** [comment]
<p>Does this no longer work?</p>
<p><code>INJECTED_ENV_VAR: ${{ secrets[github.event.inputs.injectedEnvVar] }}</code> is empty.</p>
<pre><code class="lang-auto">env:
    OTHER_SECRETS: ***
    INJECTED_ENV_VAR:
</code></pre>

**Reply by ffjdm** [reply]
And that is a pity because add-mask shows the secret in the logs when you use it.

**Reply by ZenouskaMurphy** [reply]
Input is empty for me also when trying to use the above

**Comment by rutu-sh** [comment]
With secrets, it's better if you either explicitly pass them or inherit all of them. 
Ref: 
1. [Explicitly pass secrets](https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow)
2. [Inherit all of the calling workflow's secrets](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idsecretsinherit)
